pinot-core/src/main/java/org/apache/pinot/core/common/datablock/DataBlockBuilder.java

```diff
@@ -154,6 +154,9 @@ public static RowDataBlock buildFromRows(List<Object[]> rows, DataSchema dataSch
           case OBJECT:
             setColumn(rowBuilder, byteBuffer, value);
             break;
+          case UNKNOWN:
+            setColumn(rowBuilder, byteBuffer, (Object) value);
```

Comment:
(Object) value -> (Object) null?

Reply:
The value should already be null here. but yeah, it is more clear we pas in a null. updated the code.